clang/lib/AST/ASTDumper.cpp

```diff
@@ -357,6 +357,38 @@ void ASTDumper::VisitCompoundStmt(const CompoundStmt *Node) {
   }
 }
 
+void ASTDumper::VisitWhereClause(const WhereClause *WC) {
+  if (!WC || !DumpWhereClauses)
+    return;
+
+  NodeDumper.AddChild([=] {
+    ColorScope Color(OS, ShowColors, StmtColor);
+    OS << "WhereClause";
+    
+    FactListTy Facts = WC->getFacts(); 
+    for (auto *Fact : Facts) {
+      VisitWhereClauseFact(Fact);
+      }
+  }); 
+}
+
+void ASTDumper::VisitWhereClauseFact(const WhereClauseFact *Fact) {
+  if (auto *BF = dyn_cast<BoundsDeclFact>(Fact)) {    
+    NodeDumper.AddChild([=] {
+      ColorScope Color(OS, ShowColors, DeclNameColor);
+      OS << "BoundsFact: " << BF->getVarDecl()->getQualifiedNameAsString();
+      // Visit(BF->getVarDecl());
+      Visit(BF->getBoundsExpr());
+    });
+  } else if (auto *EF = dyn_cast<EqualityOpFact>(Fact)) {
+    NodeDumper.AddChild([=] {
+      ColorScope Color(OS, ShowColors, DeclNameColor);
+      OS << "RelationalFact";
```

Comment:
Nit: "RelationalFact" can be changed to "ComparisonFact". The reason is only to align with clang internal function names. The function `isRelationalOp` returns true if the operator in question is any of <, <=, >, >=. Whereas, `isComparisonOp` returns true for <, <=, >, >=, ==, !=.

Reply:
Thanks. I will rename this.